File: src/main/java/com/flexpoker/game/command/repository/GameEventRepository.java
package com.flexpoker.game.command.repository;

import java.util.List;
import java.util.UUID;

import com.flexpoker.game.command.events.GameCreatedEvent;
import com.flexpoker.game.command.framework.GameEvent;

public interface GameEventRepository {

    List<GameEvent> fetchAll(UUID id);

    List<GameEvent> setEventVersionsAndSave(int basedOnVersion, List<GameEvent> events);

    GameCreatedEvent fetchGameCreatedEvent(UUID aggregateId);

}

File: src/main/kotlin/com/flexpoker/game/command/repository/GameEventRepository.kt
package com.flexpoker.game.command.repository

import com.flexpoker.game.command.events.GameCreatedEvent
import com.flexpoker.game.command.events.GameEvent
import java.util.UUID

interface GameEventRepository {
    fun fetchAll(id: UUID): List<GameEvent>
    fun setEventVersionsAndSave(basedOnVersion: Int, events: List<GameEvent>): List<GameEvent>
    fun fetchGameCreatedEvent(aggregateId: UUID): GameCreatedEvent?
}